### ISSUE TITLE
Invalid decoding for original UUencode charset

### ISSUE THREAD

**Issue by ecofi** [UmamiAppearance/BaseExJS#52]
Hi,

I don't know if I'm using the incorrect options but I don't get UUencode working for decoding the attached string ( see attached file).
I'm using:
`let buffer = UUencode().decode(str)`

The string contains space and I get an error message for invalid space character. The attache sample represents an encoded binary as jpg picture.

[pic_uue.txt](https://github.com/UmamiAppearance/BaseExJS/files/11518827/pic_uue.txt)

**Comment by UmamiAppearance** [comment]
The space is only part of the original charset for UUencode.

You can set this during initialization (or on the fly):
```js
const uuencode = new UUencode("original");
uuencode.decode("str");
```
I have not tested your input though. If you still have issues, just let me know.

**Comment by ecofi** [comment]
Thank you, unfortunately it is not working. Difficult to understand the problem, tested a lot.
I put a little code together for decoding to uint8array and downloading as file :
[https://jsfiddle.net/hleogr/2yuk5s34/32/](url)

I cannot say whether the input is wrong or there is a issue with the decoder.

**Comment by UmamiAppearance** [comment]
Hi. The code looks way too complicated. As far as I can see, you feed the decoder line by line, which is not the right way to do it. Where is the UUencoded string coming from? It also contains octal escape sequences. Do you have a pure sample input?

**Comment by ecofi** [comment]
Hi, indeed, the sample string is from a larger [source](https://www.sec.gov/Archives/edgar/data/1877461/000147793223001430/0001477932-23-001430.txtl)  that contains multiple uue encoded binaries. For simplification purpose, I only included one uue encoded string in the fiddle example. The matchAll regexp finds all uuencoded files and each  is feeded as ${bincontent} to the decoder (not line by line).

**Comment by UmamiAppearance** [comment]
OK. I found the problem. 
Actually it is immanent to the original uuencode charset. If you take a look at your fiddle example, line 6. There is a space missing at the very end of the line, which can easily get removed by editors or whatever. It is also missing in the original file. This causes the decoder to produce wrong results. I will work on a fix for the original version to add missing spaces if they not present (the very first character indicates the correct number of characters).

**Comment by UmamiAppearance** [comment]
Give it a try now.

**Comment by ecofi** [comment]
Hi,

First of all, thank you! I tested and for all graphics it appears to work
now. However, it is not working in other case of uuencoded binaries. I
double checked everything (including used regexp).

In some cases it is just return 3 bytes and those bytes always representing
this uint8array [ 80, 75, 3 ]. For example the first lines when just
returning those 3 bytes are:
M4$L#!!0    ( ..***@***.***   +$    0    9&]C4')O<',O87!P+GAM
M;$V./0L",1!$_\IQO;=!P4)B0-!2L+(/>QLOD&1#LD)^OCG!CVX>;QA&WPIG
M*N*I#BV&5(_C(I(/ !47BK9.7:=N')=HI6-Y #OGD<Z,STA)8*O4'J@)I9GF
***@***.***>K7A.YNJQ<&4GPZ4A!0W_<FW>J=0U[R;UEA_6\#MI7E!+ P04
M    " #CAFI6/HK7?.\    K @  $0   &1O8U!R;W!S+V-O<F4N>&ULS9+/
M3L,P#(=?!>7>.FT9AZC+96BG(2$Q"<0M2KPMHOFCQ*C=V].6K1."!^ 8^Y?/
[...]

In other cases, the CPU is running hot and decoding is not stopping or it
takes a very long time but the decoding result is not correct.
Not sure how I can help  to find the reason.



Am So., 21. Mai 2023 um 12:33 Uhr schrieb UmamiAppearance <
***@***.***>:

**Comment by UmamiAppearance** [comment]
Hi. How big are those files? Do you have a real example?

**Comment by ecofi** [comment]
HI, I attach a sample uue encoded jpg file where the CPU (16 threads) is
running hot and decoding is not stopping. The jpg files itself is only
about 300kBytes.



Am Mo., 22. Mai 2023 um 16:03 Uhr schrieb UmamiAppearance <
***@***.***>:

**Comment by UmamiAppearance** [comment]
Please continue this in this new created issue: #54

**Comment by UmamiAppearance** [comment]
Updated #54

**Comment by UmamiAppearance** [comment]
There still was an issue, when spaces where missing in last encoded line. Fixed in #56
